### PR TITLE
Updated Gem dependencies and migrate to GitHub actions

### PR DESCRIPTION
**Pull request by brototyp**

### CODE REVIEW

**Review by pstued — APPROVED** [review]
Thanks for updating!

**Comment by brototyp** [comment]
> Thanks for updating!

Thanks for the super fast review. I am somehow struggling to run the tests on circle-ci and thought we can switch over to GH Actions at the same time.

File: .github/workflows/unittest.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+on:
+  push:
+    branches: [ master, develop ]
+  pull_request:
+    branches: [ master, develop ]
+
+jobs:    
+  test:
+    name: Test
```
Comment (brototyp) [inline .github/workflows/unittest.yml:11]: I think in the future we can split up this job to be a matrix job, so we can parallelize the testing on different devices better. For now, I thought we can keep it as it was on travis.

File: Rakefile [hunk 2]
```diff
@@ -7,14 +7,14 @@ namespace :test do
 
   desc 'Run the Cauli Unit tests'
   task ios: :prepare do
-    run_tests('Cauliframework', 'iphonesimulator', 'iPhone 6', '12.2')
-    run_tests('Cauliframework', 'iphonesimulator', 'iPhone 6', '11.4')
-    run_tests('Cauliframework', 'iphonesimulator', 'iPhone 6', '10.3.1')
+    run_tests('Cauliframework', 'iphonesimulator', 'iPhone 6', '12.4')
+    run_tests('Cauliframework', 'iphonesimulator', 'iPhone 7', '13.7')
+    run_tests('Cauliframework', 'iphonesimulator', 'iPhone 8', '14.4')
```
Comment (brototyp) [inline Rakefile:12]: Simulators for iOS 10 and 11 aren't provided by Github.

**Review by pstued — APPROVED** [review]

**Pull request merged**